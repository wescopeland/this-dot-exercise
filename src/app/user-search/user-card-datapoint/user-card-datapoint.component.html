<div class="tde-github-user-card-datapoint">
  <mat-icon [ngClass]="{ 'mr-1': !hyperlink }">{{ icon }}</mat-icon>
  <span *ngIf="!hyperlink">{{ text }}</span>
  <a 
    *ngIf="hyperlink"
    target="_blank"
    rel="noopener"
    [href]="hyperlink">
    {{ text }}
  </a>
</div>