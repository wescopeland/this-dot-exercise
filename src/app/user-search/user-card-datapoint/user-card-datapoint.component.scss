.tde-github-user-card-datapoint {
  .mat-icon {
    vertical-align: bottom;
    color: rgba(0, 0, 0, 0.54);
  }

  span,
  a {
    font-size: 14px;
  }
}
