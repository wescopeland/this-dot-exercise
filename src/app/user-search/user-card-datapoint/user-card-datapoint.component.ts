import { Component, Input, OnInit } from '@angular/core';

@Component({
  selector: 'tde-github-user-card-datapoint',
  templateUrl: './user-card-datapoint.component.html',
  styleUrls: ['./user-card-datapoint.component.scss']
})
export class UserCardDatapointComponent implements OnInit {
  @Input() icon: string;
  @Input() hyperlink: string;
  @Input() text: string;

  constructor() {}

  ngOnInit() {}
}
