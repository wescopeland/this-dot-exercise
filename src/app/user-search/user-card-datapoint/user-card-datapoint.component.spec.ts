import { Spectator } from '@netbasal/spectator';
import { createTestComponentFactory } from '@netbasal/spectator/jest';
import { MatIconModule } from '@angular/material/icon';

import { UserCardDatapointComponent } from './user-card-datapoint.component';

describe('Component: UserCardDatapointComponent', () => {
  let spectator: Spectator<UserCardDatapointComponent>;
  const createComponent = createTestComponentFactory<
    UserCardDatapointComponent
  >({
    component: UserCardDatapointComponent,
    detectChanges: false,
    declarations: [],
    imports: [MatIconModule],
    componentProviders: []
  });

  beforeEach(() => {
    spectator = createComponent();
  });

  it('exists', () => {
    expect(spectator.component).toBeDefined();
  });

  it('can render a datapoint without a hyperlink', () => {
    spectator.component.icon = 'close';
    spectator.component.text = 'no hyperlink';
    spectator.detectChanges();

    expect(spectator.fixture).toMatchSnapshot();
    expect(spectator.query('span')).toHaveText('no hyperlink');
    expect(spectator.query('a')).not.toBeTruthy();
  });

  it('can render a datapoint with a hyperlink', () => {
    spectator.component.icon = 'close';
    spectator.component.text = 'hyperlink';
    spectator.component.hyperlink = 'https://abc.xyz';
    spectator.detectChanges();

    expect(spectator.fixture).toMatchSnapshot();
    expect(spectator.query('a')).toBeTruthy();
    expect(spectator.query('a')).toHaveAttribute('href', 'https://abc.xyz');
  });
});
