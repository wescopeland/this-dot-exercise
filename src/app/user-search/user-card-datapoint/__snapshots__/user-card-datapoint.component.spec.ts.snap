// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserCardDatapointComponent can render a datapoint with a hyperlink 1`] = `
<tde-github-user-card-datapoint
  hyperlink={[Function String]}
  icon={[Function String]}
  text={[Function String]}
>
  <div
    class="tde-github-user-card-datapoint"
  >
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate material-icons mat-icon-no-color"
      ng-reflect-ng-class="[object Object]"
      role="img"
    >
      close
    </mat-icon>
    
    
    <a
      href="https://abc.xyz"
      rel="noopener"
      target="_blank"
    >
       hyperlink 
    </a>
  </div>
</tde-github-user-card-datapoint>
`;

exports[`Component: UserCardDatapointComponent can render a datapoint without a hyperlink 1`] = `
<tde-github-user-card-datapoint
  icon={[Function String]}
  text={[Function String]}
>
  <div
    class="tde-github-user-card-datapoint"
  >
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate mr-1 material-icons mat-icon-no-color"
      ng-reflect-ng-class="[object Object]"
      role="img"
    >
      close
    </mat-icon>
    
    <span>
      no hyperlink
    </span>
    
  </div>
</tde-github-user-card-datapoint>
`;
